# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/nsClipboardProxy.cpp
# Commit: 8ab38c9b4f25
# Full Hash: 8ab38c9b4f25ed4437804b708cb3be993c522400
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2022-08-26 21:48:35
# Regressor Bug: 1781129
# File Overlap Count: 1
# Description:
#   Bug 1781129 - Part 1: Use BigBuffer for IPCDataTransfer, r=edgar
#   
#   The IPCDataTransfer type is used to transfer Clipboard/Drag & Drop payloads
#   over IPC to allow them to be written to or read from the relevant system
#   interfaces. Previously, the system which was used was somewhat complex, and
# ==============================================================================

diff -r a9c593b5a54a -r 8ab38c9b4f25 widget/nsClipboardProxy.cpp
--- a/widget/nsClipboardProxy.cpp	Fri Aug 26 15:50:18 2022 +0000
+++ b/widget/nsClipboardProxy.cpp	Fri Aug 26 16:08:05 2022 +0000
@@ -33,8 +33,9 @@
   nsCOMPtr<nsIPrincipal> requestingPrincipal =
       aTransferable->GetRequestingPrincipal();
   nsContentPolicyType contentPolicyType = aTransferable->GetContentPolicyType();
-  child->SendSetClipboard(ipcDataTransfer, isPrivateData, requestingPrincipal,
-                          contentPolicyType, aWhichClipboard);
+  child->SendSetClipboard(std::move(ipcDataTransfer), isPrivateData,
+                          requestingPrincipal, contentPolicyType,
+                          aWhichClipboard);
 
   return NS_OK;
 }
@@ -49,8 +50,7 @@
   ContentChild::GetSingleton()->SendGetClipboard(types, aWhichClipboard,
                                                  &dataTransfer);
   return nsContentUtils::IPCTransferableToTransferable(
-      dataTransfer, false /* aAddDataFlavor */, aTransferable,
-      ContentChild::GetSingleton());
+      dataTransfer, false /* aAddDataFlavor */, aTransferable);
 }
 
 NS_IMETHODIMP
@@ -129,8 +129,7 @@
           /* resolve */
           [promise, transferable](const IPCDataTransfer& ipcDataTransfer) {
             nsresult rv = nsContentUtils::IPCTransferableToTransferable(
-                ipcDataTransfer, false /* aAddDataFlavor */, transferable,
-                ContentChild::GetSingleton());
+                ipcDataTransfer, false /* aAddDataFlavor */, transferable);
             if (NS_FAILED(rv)) {
               promise->Reject(rv, __func__);
               return;